File: kerbianbridge/android/ErrorReporter.java
package kerbianbridge.android;

import org.json.JSONObject;

public class ErrorReporter {
    public static NativeEventSink eventSink;

    public static void report(String errorType, String message, JSONObject details) {
        try {
            if (eventSink != null) {
                JSONObject errorMsg = new JSONObject();
                errorMsg.put("type", "error");
                errorMsg.put("error_type", errorType);
                errorMsg.put("message", message);
                errorMsg.put("details", details != null ? details : new JSONObject());
                eventSink.send(errorMsg.toString());
            }
        } catch (Exception e) {
            System.err.println("Bridge error: " + errorType + " " + message);
            e.printStackTrace();
        }
    }
}

File: kerbianbridge/ios/ErrorReporter.m
#import "ErrorReporter.h"

static id _eventSink = nil;

@implementation ErrorReporter
+ (void)setEventSink:(id)sink { _eventSink = sink; }
+ (void)report:(NSString *)errorType message:(NSString *)message details:(NSDictionary *)details {
    if (_eventSink && [_eventSink respondsToSelector:@selector(send:)]) {
        NSDictionary *err = @{
            @"type": @"error",
            @"error_type": errorType,
            @"message": message,
            @"details": details ?: @{}
        };
        NSData *jsonData = [NSJSONSerialization dataWithJSONObject:err options:0 error:nil];
        NSString *jsonLine = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
        [_eventSink send:jsonLine];
    }
}
@end